Sources/SwiftParser/Patterns.swift

```diff
@@ -354,6 +354,7 @@ extension Parser.Lookahead {
         && !self.at(.keyword(.__owned))
         && !self.at(.keyword(.borrowing))
         && !self.at(.keyword(.consuming))
+        && !self.at(.keyword(.resultDependsOn))
```

Comment:
Similar here, you should be checking that the experimental language features is enabled (`experimentalFeatures.contains(.typedThrows)`).